src/psyclone/tests/gocean1p0_transformations_test.py

```diff
@@ -1184,3 +1184,196 @@ def test_module_inline_warning_no_change():
     kern_call = schedule.children[0].children[0].children[0]
     inline_trans = KernelModuleInlineTrans()
     _, _ = inline_trans.apply(kern_call, inline=False)
+
+
+def test_loop_swap_correct():
+    ''' Testing correct loop swapping transform. Esp. try first, middle, and
+    last invokes to make sure the inserting of the inner loop happens at
+    the right place.'''
+
+    psy, invoke = get_invoke("test27_loop_swap.f90", 0)
+    schedule = invoke.schedule
+
+    # First make sure to throw an early error if the source file
+    # single_invoke_two_kernels.f90 should have been changed
+    assert str(psy.gen) == """  MODULE psy_test27_loop_swap
```

Comment:
I wouldn't bother testing against the whole MODULE here - just the body of the subroutine. And yes, we are picky about white space and long lines :-) See some of the dynamo tests for examples.